testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,711 @@
+                allFields.addAll(Arrays.asList(declaredFields));
+                allFields.addAll(Arrays.asList(fields));
+            } catch (NoClassDefFoundError e) {
+                e.printStackTrace();
+                continue;
+            } catch (Exception exception) {
+                exception.printStackTrace();
+                continue;
+            }
+            // prepare for the subxml fold
+
+            for (Field f : allFields) {
+                String fieldName = getFieldFQN(f);
+
+                if (ignores.contains(fieldName)) {
+                    continue;
+                }
+
+                // if a field is final and has a primitive type there's no point to capture it.
+                if (Modifier.isStatic(f.getModifiers())
+                    && !(Modifier.isFinal(f.getModifiers()) &&  f.getType().isPrimitive())) {
+                    try {
+                        if (shouldCapture(f)) {
+                            allFieldName.add(fieldName);
+                            f.setAccessible(true);
+
+                            Object instance;
+                            try {
+                                instance = f.get(null);
+                            } catch (NoClassDefFoundError ncdfe) {
+                                instance = null;
+                                ncdfe.printStackTrace();
+                            }
+                            if (instance instanceof ThreadLocal) {
+                                instance = ((ThreadLocal)instance).get();
+                            }
+                            LinkedHashMap<String, Object> nameToInstance_temp = new LinkedHashMap<String, Object>();
+                            nameToInstance_temp.put(fieldName, instance);
+
+                            dirty = false;
+                            serializeRoots(nameToInstance_temp);
+                            if (!dirty) {
+                                nameToInstance.put(fieldName, instance);
+
+                                String ob4field = serializeOBs(instance);
+                                writer = new PrintWriter(subxmlDir + "/" + fieldName + ".xml", "UTF-8");
+                                writer.println(ob4field);
+                                writer.close();
+                            }
+                        }
+                    } catch (OutOfMemoryError ofme) {
+                        ofme.printStackTrace();
+                        continue;
+                    } catch (NoClassDefFoundError ncdfe) {
+                        ncdfe.printStackTrace();
+                        continue;
+                    } catch (Exception exception) {
+                        exception.printStackTrace();
+                        continue;
+                    }
+                }
+            }
+        }
+
+        String state = Configuration.config().getProperty("statecapture.state");
+        if (state.equals("passing_order")) {
```

Comment:
Can we explicitly pass in the name of the files to write to instead of needing to distinguish based on passing or failing order? In general, the logic for capturing state does not care whether overall test order passes or fails. Imagine that the state-capturing logic is not specific to just test-order things, that it can be applied to a variety of other tasks.